optimum/intel/neural_compressor/modeling_base.py

```diff
@@ -0,0 +1,512 @@
+                        cache_dir = model_kwargs.get("cache_dir", None)
+                        if cache_dir is None:
+                            cache_dir = TRANSFORMERS_CACHE
+                        if isinstance(cache_dir, Path):
+                            cache_dir = str(cache_dir)
+                        try:
+                            state_dict_path = hf_hub_download(
+                                repo_id=model_id,
+                                filename=file_name,
+                                revision=revision,
+                                cache_dir=cache_dir,
+                                local_files_only=local_files_only,
+                            )
+                        except EnvironmentError as err:
+                            logger.error(err)
+                            msg = (
+                                f"Can't load config for '{model_id}'. Make sure that:\n\n"
+                                f"-'{model_id}' is a correct model identifier listed on 'https://huggingface.co/models'\n\n"
+                                f"-or '{model_id}' is a correct path to a directory containing a {file_name} file\n\n"
+                            )
+
+                            if revision is not None:
+                                msg += (
+                                    f"- or '{revision}' is a valid git identifier (branch name, a tag name, or a commit id) that "
+                                    f"exists for this model name as listed on its model page on 'https://huggingface.co/models'\n\n"
+                                )
+
+                            raise EnvironmentError(msg)
+
+                msg = None
+                try:
+                    inc_config = INCConfig.from_pretrained(model_id)
+                    if not is_torch_version("==", inc_config.torch_version):
+                        msg = f"Quantized model was obtained with torch version {inc_config.torch_version} but {_torch_version} was found."
+                        logger.warning(f"{msg}")
+                except Exception:
+                    logger.info("Couldn't verify torch version.")
+
+                # Load the state dictionary of the model to verify whether the model is quantized or not
+                state_dict = torch.load(state_dict_path, map_location="cpu")
+
+                if "best_configure" in state_dict and state_dict["best_configure"] is not None:
+                    try:
+                        model = load(state_dict_path, model)
+                    except Exception as e:
+                        if msg is not None:
+                            e.args += (msg,)
+                        raise
+
+            model.eval()
+
+        return cls(
+            model,
+            config=config,
+            model_save_dir=model_save_dir,
+            use_cache=use_cache,
+            **kwargs,
+        )
+
+    @classmethod
+    def _from_transformers(
```

Comment:
Mostly already integrated in [`TSModelForCausalLM._from_transformers`](https://github.com/huggingface/optimum-intel/blob/c48f06ad3303df8f1c530db6f953c23f6e71553c/optimum/intel/generation/modeling.py#L235)

Reply:
There is a little difference with [TSModelForCausalLM._from_transformers](https://github.com/huggingface/optimum-intel/blob/c48f06ad3303df8f1c530db6f953c23f6e71553c/optimum/intel/generation/modeling.py#L235), in this function, we support load quantized model with INC and then jit trace the model here.